include/check_out.php

```
@@ -51,36 +51,32 @@
         $cmsmain->assign('formelements', $formdata);
         $cmsmain->assign('formbuttons', $formbuttons);
     } else {
-        if ($_POST['do']) {
-            switch ($_POST['do']) {
-                case 'delete':
-                    $ids = explode(',', $_POST['ids']);
-                    list($success, $message, $redirect) = listDelete($table, $ids);
-
-                    break;
-                case 'hide':
-                    $ids = explode(',', $_POST['ids']);
-                    list($success, $message, $redirect) = listShowHide($table, $ids, 0);
-
-                    break;
-                case 'show':
-                    $ids = explode(',', $_POST['ids']);
-                    list($success, $message, $redirect) = listShowHide($table, $ids, 1);
-
-                    break;
+        if ('delete' == $_POST['do']) {
+            $ids = explode(',', $_POST['ids']);
+            // check if person is allowed to delete transaction
+            foreach ($ids as $id) {
+                verify_campaccess_people(db_value('SELECT people_id FROM transactions WHER id=:id', ['id' => $id]));
             }
-
+            list($success, $message, $redirect) = listDelete($table, $ids);
             $return = ['success' => $success, 'message' => $message, 'redirect' => false, 'action' => "$('#field_people_id').trigger('change')"];
 
             echo json_encode($return);
             die();
         }
 
+        // verify acces if data of a person is requested
+        verify_campaccess_people($_POST['people_id']);
+        verify_deletedrecord('people', $_POST['people_id']);
+
+        // set camp depending on people_id
+        $camp = db_row('
```

I know this shouldn't happen, but I'm just wondering what happens if the query fails and returns null or something...should we maybe check whether the $camp resultset has a value? And if it doesn't then silently skip the next steps or maybe return some error message?

---

Camp should not be null in any case. If so it will throw the same error as before and in line 68 there is already a check if the user has access.